src/tools/tidy/src/iter_header.rs

```diff
@@ -0,0 +1,31 @@
+const COMMENT: &str = "//@";
+
+/// A header line, like `//@name: value` consists of the prefix `//@` and the directive
+/// `name: value`. It is also possibly revisioned, e.g. `//@[revision] name: value`.
+pub(crate) struct HeaderLine<'ln> {
+    pub(crate) revision: Option<&'ln str>,
+    pub(crate) directive: &'ln str,
+}
+
+/// Iterate through compiletest headers in a test contents.
+///
+/// Adjusted from compiletest/src/header.rs.
+pub(crate) fn iter_header<'ln>(contents: &'ln str, it: &mut dyn FnMut(HeaderLine<'ln>)) {
+    for ln in contents.lines() {
+        let ln = ln.trim();
+        if ln.starts_with(COMMENT) && ln[COMMENT.len()..].trim_start().starts_with('[') {
+            if let Some(close_brace) = ln.find(']') {
+                let open_brace = ln.find('[').unwrap();
+                let revision = &ln[open_brace + 1..close_brace];
```

Comment:
```suggestion
        if let Some(comment) = ln.strip_prefix(COMMENT)
            && let Some(comment) = comment.trim_start().strip_prefix('[')
            && let Some(revision) = comment.strip_suffix(']')
        {
```

Reply:
This requires `let_chains` right? Can tidy use a nightly feature?

Reply:
Ah right, I guess not. However, I'd still use `strip_{prefix,suffix}` here over unwrapping and manual slice indexing even without let-chains by just using nested if-lets.

Reply:
You also use let-else with `continue` to avoid rightward drift.